### DISCUSSION TITLE
Federation directive

### DISCUSSION THREAD

**Discussion by aldis-ameriks** [strapi/strapi#7195] [Legacy Topics]
I want to extend a Strapi managed type from separate graphql service both running behind federated gateway. According to federation docs, I would need to add `@key` directive on both ends. Is there a way to add it for the Strapi types?

Thanks.

**Comment by derrickmehaffy** [comment]
Hey @alexandrebodin and @Aurelsicoko do you have suggestions on this topic.

**Reply by alexandrebodin** [reply]
You can use the extend directive in custom schemas but not sure if this would work

**Reply by aldis-ameriks** [reply]
To provide a little more context, I'm having this use case where the "main data" source comes from service "A", but I want to sprinkle the "main data" with some texts and images that will be visible in UI. So I link the "main data" from service "A" with Strapi cms (as a separate service) through a common "name" field.
I was hoping of achieving it by using federation and linking the data together via the @key directive. But I'd rather not overcomplicate the setup with custom resolvers/schema on Strapi end. So at the moment, I'm stiching the data together in frontend manually.

It would be really awesome if this type of use case was supported. But I do understand that it's a bit niche and majority of users won't benefit from it.
